src/raft.rs

```diff
@@ -1993,12 +1993,13 @@ impl<T: Storage> Raft<T> {
     fn check_quorum_active(&mut self) -> bool {
         let self_id = self.id;
         let mut act = 0;
+        let learners = self.prs().learner_ids().clone();
```

Comment:
The clone here seems the best way, as much as I hate it. Since we use `.mut_prs()` and iterate we have a mutable borrow out for `self`. 

We copy this `HashSet<u64>` which should only have learner _ids() in it, which should be not unreasonably large. An alternative is we can clone `voter_ids()` which should always be reasonable even if there are many learners.

Reply:
You can use `take_prs` to get around the lifetime check just as other places already did.

Reply:
I always feel so dirty doing this. :rofl: But yes you're right.